core/templates/pages/exploration-editor-page/services/router.service.ts

```diff
@@ -221,9 +221,16 @@ export class RouterService {
   }
 
   isLocationSetToNonStateEditorTab(): boolean {
-    let currentPath = (
-      '/' +
-      this.windowRef.nativeWindow.location.hash.split('#')[1].split('/')[1]);
+    let hashParts = this.windowRef.nativeWindow.location.hash ?
+      this.windowRef.nativeWindow.location.hash.split('#') : [];
+    let currentPath = '/';
+
+    if (hashParts.length > 1) {
+      let pathParts = hashParts[1].split('/');
+      if (pathParts.length > 1) {
+        currentPath += pathParts[1];
+      }
+    }
```

Comment:
Is it possible to compress this using `Nullish coalescing` and `optional chaining`?
<code>currentPath = ('/' + (this.windowRef.nativeWindow.location.hash?.split('#').get(1)?.split('/').get(1)) ?? '')</code>

Reply:
Done